web/ui/react-app/src/pages/targets/ScrapePoolList2.tsx

```diff
@@ -0,0 +1,131 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { ScrapePoolContent } from './ScrapePoolContent';
+import Filter, { Expanded, FilterData } from './Filter';
+import { useLocalStorage } from '../../hooks/useLocalStorage';
+import styles from './ScrapePoolPanel.module.css';
+import { ToggleMoreLess } from '../../components/ToggleMoreLess';
+
+interface ScrapePoolListProps {
+  activeTargets: Target[];
+}
+
+const kvSearch = new KVSearch({
+  shouldSort: true,
+  indexedKeys: ['labels', 'scrapePool', ['labels', /.*/]],
+});
+
+interface PanelProps {
+  scrapePool: string;
+  targetGroup: ScrapePool;
+  expanded: boolean;
+  toggleExpanded: () => void;
+}
+
+function ScrapePoolPanel(props: PanelProps): JSX.Element {
+  const modifier = props.targetGroup.upCount < props.targetGroup.targets.length ? 'danger' : 'normal';
+  const id = `pool-${props.scrapePool}`;
+  const anchorProps = {
+    href: `#${id}`,
+    id,
+  };
+  return (
+    <div>
+      <ToggleMoreLess event={props.toggleExpanded} showMore={props.expanded}>
+        <a className={styles[modifier]} {...anchorProps}>
+          {`${props.scrapePool} (${props.targetGroup.upCount}/${props.targetGroup.targets.length} up)`}
+        </a>
+      </ToggleMoreLess>
+      <Collapse isOpen={props.expanded}>
+        <ScrapePoolContent targets={props.targetGroup.targets} />
+      </Collapse>
+    </div>
+  );
+}
+
+// ScrapePoolListContent is taking care of every possible filter
+function ScrapePoolListContent(props: ScrapePoolListProps): JSX.Element {
+  const initialPoolList = groupTargets(props.activeTargets);
+  const [poolList, setPoolList] = useState<ScrapePools>(initialPoolList);
+  const initialFilter: FilterData = {
+    showHealthy: true,
+    showUnhealthy: true,
+  };
+  const [filter, setFilter] = useLocalStorage('targets-page-filter', initialFilter);
+
+  const initialExpanded: Expanded = Object.keys(initialPoolList).reduce(
+    (acc: { [scrapePool: string]: boolean }, scrapePool: string) => ({
+      ...acc,
+      [scrapePool]: true,
+    }),
+    {}
+  );
+  const [expanded, setExpanded] = useLocalStorage('targets-page-expansion-state', initialExpanded);
+  const { showHealthy, showUnhealthy } = filter;
+
+  const handleSearchChange = (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+    if (e.target.value !== '') {
+      const result = kvSearch.filter(e.target.value.trim(), props.activeTargets);
+      setPoolList(
+        groupTargets(
+          result.map((value) => {
+            return value.original as unknown as Target;
+          })
+        )
+      );
+    } else {
+      setPoolList(initialPoolList);
+    }
+  };
+  return (
+    <>
+      <Filter filter={filter} setFilter={setFilter} expanded={expanded} setExpanded={setExpanded} />
+      <Container>
+        <InputGroup>
+          <InputGroupText>{<FontAwesomeIcon icon={faSearch} />}</InputGroupText>
```

Comment:
Could you wrap this in a prepend element like in https://github.com/prometheus/prometheus/blob/a961062c37f1675d5925d8b74ddda0a6322ed82d/web/ui/react-app/src/pages/graph/ExpressionInput.tsx#L219-L223, so it visually gets merged with the input?